plugins/modules/ec2_asg_instance_refreshes_info.py

```diff
@@ -0,0 +1,229 @@
+    returned: success
+    type: int
+    sample: 100
+instances_to_update:
+    description: num. of instance to update
+    returned: success
+    type: int
+    sample: 5
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def find_asg_instance_refreshes(conn, module):
+    """
+    Args:
+        conn (boto3.AutoScaling.Client): Valid Boto3 ASG client.
+        name (str): Mandatory name of the ASG you are looking for.
+        ids (dict): Optional list of ASG Instace Refresh IDs
+        max_records (int): Optional number of max records to return
+        next_token (str): Optional NextToken from previous call
+
+    Basic Usage:
+        >>> name = 'some-asg'
+        >>> conn = boto3.client('autoscaling', region_name='us-west-2')
+        >>> results = find_asg_instance_refreshes(name, conn)
+
+    Returns:
+        Dict
+        {
+            'instance_refreshes': [
+                    {
+                        'instance_refresh_id': '6507a3e5-4950-4503-8978-e9f2636efc09',
+                        'auto_scaling_group_name': 'ansible-test-hermes-63642726-asg',
+                        'status': 'Cancelled',
+                        'status_reason': 'Cancelled due to user request.',
+                        'start_time': '2021-02-04T03:39:40+00:00',
+                        'end_time': '2021-02-04T03:41:18+00:00',
+                        'percentage_complete': 0,
+                        'instances_to_update': 1
+                    }
+            ],
+            'next_token': 'string'
+        }
+        """
+
+    asg_name = module.params.get('name')
+    asg_ids = module.params.get('ids')
+    asg_next_token = module.params.get('next_token')
+    asg_max_records = module.params.get('max_records')
+
+    args = {}
+    args['AutoScalingGroupName'] = asg_name
+    if asg_ids:
+        args['InstanceRefreshIds'] = asg_ids
+    if asg_next_token:
+        args['NextToken'] = asg_next_token
+    if asg_max_records:
+        args['MaxRecords'] = asg_max_records
+
+    try:
+        instance_refreshes_result = {}
+        response = conn.describe_instance_refreshes(aws_retry=True, **args)
+        if 'InstanceRefreshes' in response:
+            instance_refreshes_dict = dict(
+                instance_refreshes=response['InstanceRefreshes'], next_token=response.get('next_token', ''))
+            instance_refreshes_result = camel_dict_to_snake_dict(
+                instance_refreshes_dict)
+
+        while 'NextToken' in response:
+            args['NextToken'] = response['NextToken']
+            response = conn.describe_instance_refreshes(aws_retry=True, **args)
```

Comment:
The aws_retry part might be redundant if already set on the initial setup of the client